compiler/rustc_privacy/src/lib.rs

```diff
@@ -1922,8 +1985,10 @@ impl<'tcx> PrivateItemsInPublicInterfacesChecker<'tcx> {
         def_id: LocalDefId,
         assoc_item_kind: AssocItemKind,
         vis: ty::Visibility,
+        effective_vis: Option<EffectiveVisibility>,
     ) {
-        let mut check = self.check(def_id, vis);
+        self.check_unnameable(def_id, self.get(def_id));
```

Comment:
```suggestion
        if is_assoc_ty { self.check_unnameable(def_id, self.get(def_id)); }
```
The `unnameable_types` lint should probably only be reported for *types* (or traits).